frontend/src/pages/DetailPage.tsx

```diff
@@ -1,15 +1,386 @@
+  const TIME_API_KEY = `${import.meta.env.VITE_TIME_API}`;
+  const [getLat, setLat] = useState('');
+  const [getLon, setLon] = useState('');
+  const [getDisplayedDate, setDisplayedDate] = useState('');
+  const [getDisplayedTime, setDisplayedTime] = useState('');
+  const [getTime, setTime] = useState<TimeResponse | null>(null);
+  const [getTemp, setTemp] = useState('');
+  const [getFeelsLikeTemp, setFeelsLikeTemp] = useState('');
+  const [getHumidity, setHumidity] = useState('');
+  const [getMaxTemp, setMaxTemp] = useState('');
+  const [getMinTemp, setMinTemp] = useState('');
+  const [getDescription, setDescription] = useState('');
+  const [getWeather, setWeather] = useState([]);
+  const [firstExchange, setFirstExchange] = useState('');
+  const [secondExchange, setSecondExchange] = useState('');
+  const [currencyAmount, setCurrencyAmount] = useState(0);
+  const [currencyResult1, setCurrencyResult1] = useState(0);
+  const [currencyResult2, setCurrencyResult2] = useState(0);
+  const [convertingCurrency, setConvertingCurrency] = useState<CurrencyResponse | null>(null);
+  const [exchangeName, setExchangeName] = useState('');
+
+  // on region
+  useEffect(() => {
+    if (region) {
+      getLatLong();
+    }
+  }, [region]);
+
+  // on lat/lon
+  useEffect(() => {
+    if (getLat && getLon) {
+      console.log('Latitude:', getLat);
+      console.log('Longitude:', getLon);
+      getCurrentTime();
+      getWeatherData();
+    }
+  }, [getLat]);
+
+  // on Time
+  useEffect(() => {
+    if (getTime && getTime.localTime) {
+      console.log(getTime);
+      console.log(getTime.localTime);
+      setDisplayedDate(getTime.localTime.substring(0, 10));
+      setDisplayedTime(getTime.localTime.substring(11, 19));
+    }
+  }, [getTime]);
+
+  // on Weather
+  useEffect(() => {
+    if (getWeather.main) {
+      console.log(getWeather);
+      setTemp(getWeather.main.temp);
+      setMaxTemp(getWeather.main.temp_max);
+      setMinTemp(getWeather.main.temp_min);
+      setFeelsLikeTemp(getWeather.main.feels_like);
+      setHumidity(getWeather.main.humidity);
+      setDescription(getWeather.weather[0].description);
+    }
+  }, [getWeather]);
+
+  useEffect(() => {
+    if (convertingCurrency) {
+      let result = 0;
+      let currenyRates = convertingCurrency.rates;
+      if (currenyRates) {
+        console.log(currenyRates);
+        if (exchangeName == 'normal') {
+          result = currencyAmount * currenyRates[secondExchange];
+          let formattedResult = result.toFixed(2);
+          setCurrencyResult2(formattedResult);
+          setCurrencyResult1(currencyAmount);
+        } else if (exchangeName == 'reverse') {
+          result = currencyAmount * currenyRates[firstExchange];
+          let formattedResult = result.toFixed(2);
+          setCurrencyResult2(currencyAmount);
+          setCurrencyResult1(formattedResult);
+        }
+      }
+    }
+  }, [convertingCurrency]);
+
+  const getLatLong = async () => {
```

Comment:
you could move these functions into there own file in `src/utils` so the code here gets easier to read.